presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2802,13 +2802,25 @@ private static TableFinisher createTableFinisher(Session session, TableFinishNod
         WriterTarget target = node.getTarget();
         return (fragments, statistics) -> {
             if (target instanceof CreateHandle) {
-                return metadata.finishCreateTable(session, ((CreateHandle) target).getHandle(), fragments, statistics);
+                CreateHandle createHandle = (CreateHandle) target;
+                if (metadata.isWritingStagingFilesEnabled(session)) {
+                    metadata.commitStagingFiles(session, createHandle.getSchemaTableName(), fragments);
```

Comment:
To be honest, it looks weird. Why do we need an additional method in the SPI, if we calling them one after the other? 

e.g.: `commitStagingFiles` -> `finishCreateTable`.

Why `finishCreateTable`/`finishInsert` cannot handle that inside?